cns/cnsclient/httpapi/client.go

```diff
@@ -12,32 +14,22 @@ type Client struct {
 
 // CreateOrUpdateNC updates cns state
 func (client *Client) CreateOrUpdateNC(ncRequest *cns.CreateNetworkContainerRequest) error {
-	// var (
-	// 	ipConfigsToAdd []*cns.ContainerIPConfigState
-	// )
-
-	// //Lock to read ipconfigs
-	// client.RestService.Lock()
-
-	// //Only add ipconfigs that don't exist in cns state already
-	// for _, ipConfig := range ipConfigs {
-	// 	if _, ok := client.RestService.PodIPConfigState[ipConfig.ID]; !ok {
-	// 		ipConfig.State = cns.Available
-	// 		ipConfigsToAdd = append(ipConfigsToAdd, ipConfig)
-	// 	}
-	// }
-
-	// client.RestService.Unlock()
-	// leave empty
-	return nil //client.RestService.AddIPConfigsToState(ipConfigsToAdd)
+	returnCode := client.RestService.CreateOrUpdateNetworkContainerInternal(*ncRequest)
+
+	if returnCode != 0 {
+		return fmt.Errorf("Failed to Create NC request: %+v, errorCode: %d", *ncRequest, returnCode)
+	}
+
+	return nil
 }
 
 // InitCNSState initializes cns state
-func (client *Client) InitCNSState(ncRequest *cns.CreateNetworkContainerRequest, podInfoByIP map[string]*cns.KubernetesPodInfo) error {
```

Comment:
You'll also need to change the name in apiclient.go and in crdrequestconroller_test.go if you change here

Reply:
As well as the map type